### PR TITLE
Fix push of Levels at or below zero

### LINKED ISSUE #313: ETABS_Toolkit: Level with elevation 0 doesn't push, returns irrelevant error.
#### Description:
Attempting to push a list of levels including one at elevation = 0 fails for all levels, the push returns the error 'can only push levels to an empty model' instead of 'could not push level with elevation 0' so it's very difficult to troubleshoot.

#### Steps to reproduce:
attempt to push a list of levels, one of which is at elevation 0.

#### Expected behaviour:
preferably handle levels with elevation =  0 by renaming the base level or otherwise dealing with it automatically. Alternately, throw a useful error letting the user know to cull the 0th level.

### PR DESCRIPTION
**Pull request by kThorsager**
### Issues addressed by this PR

 Closes #313 

 

 ### Test files

https://burohappold.sharepoint.com/:u:/s/BHoM/Ed6BQd-J0YVFpOvCapmRme4B78umJf1o-av4JswmxqT9YA?e=BTnKR0

 ### Changelog

- Change made so that the level provided with the lowest elevation is always considered the base level. No longer inserting base at level 0.

 ### Additional comments

### CODE REVIEW

File: Etabs_Adapter/CRUD/Create/Level.cs [hunk 1]
```diff
@@ -55,14 +55,33 @@ private bool CreateCollection(IEnumerable<Level> levels)
...
+#if Debug16 || Release16
             double[] elevations = new double[count + 1];
 
             for (int i = 0; i < count; i++)
             {
                 elevations[i + 1] = levelList[i].Elevation;
             }
 
-            double[] heights = new double[count];   //Heights empty, set by elevations
+            if (elevations.Any(x => x <= 0))
```
Comment (IsakNaslundBh) [inline Etabs_Adapter/CRUD/Create/Level.cs:68]: This check will always be true, as 0 level is inserted to the elevation (by not being assigned) and from memory, I think this is required for the API to play ball. Another check will be needed for etabs 2016.

Comment (kThorsager) [reply Etabs_Adapter/CRUD/Create/Level.cs:68]: Right, should probably mention the difficulties in testing due to not having ETABS16

**Comment by kThorsager** [comment]
@IsakNaslundBh Think I've fixed it, but please do test (still no ETABS16 on my pc)

**Comment by peterjamesnugent** [comment]
@JosefTaylor to review @kThorsager changes.

**Comment by JosefTaylor** [comment]
ETABS 16 now gives an error for negative or zero elevation levels, however ETABS 16 fails for positive levels (with no negative or zero levels pushed)

ETABS 17 does not succeed in pushing negative or zero elevation levels, and does not give a useful error ("levels can only be pushed to an empty model"). Pushing levels which are positive and non zero is successful.

**Comment by IsakNaslundBh** [comment]
> @IsakNaslundBh do you mind taking over this and then @JosefTaylor can review and I will push it to Sprint 3.

Thanks @peterjamesnugent , yes, happy to take this on and have Josef review!

**Comment by IsakNaslundBh** [comment]
@JosefTaylor found a way to push negative levels to Etabs 2016 as well, have pushed that up. Had to make a slight exception to what we did before, but realised, that we might want to have that exception as a default potentially for all cases.

Lets catch up quickly today and discuss before merging this PR.

**Comment by IsakNaslundBh** [comment]
ok @JosefTaylor this should now be ready for review, following the strategy agreed. Will update the changelog in the description as well

**Comment by IsakNaslundBh** [comment]
Thanks @alelom

**Pull request merged**